app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -409,21 +409,20 @@ def save_labels_inventory(entity, hashes, target = nil)
     store_ids_for_new_records(entity.labels, hashes, [:section, :name])
   end
 
-  def save_labels_and_tags_inventory(entity, hashes, target = nil)
-    save_labels_inventory(entity, hashes, target)
-    save_tags_generated_from_labels(entity, hashes, target)
-  end
+  def save_tags_inventory(entity, hashes, _target = nil)
+    return if hashes.nil?
 
-  def save_tags_generated_from_labels(entity, hashes, _target = nil)
-    labels = hashes.collect do |label_hash|
-      OpenStruct.new(:resource_type => entity.class.base_class.name,
-                     :name          => label_hash[:name],
-                     :value         => label_hash[:value])
+    begin
+      mapped_tags = hashes.map { |tag_hash| ContainerLabelTagMapping.find_or_create_tag(tag_hash) }
+    rescue => err
+      raise if EmsRefresh.debug_failures
```

Comment:
@cben I think this is needed to see the backtrace, but if you're printing it always (see two lines below), do we still need this?